python_modules/dagster/dagster/check/__init__.py

```diff
@@ -276,6 +276,10 @@ def opt_str_param(obj: Any, param_name: str, default: str = None) -> Optional[st
         raise _param_type_mismatch_exception(obj, str, param_name)
     return default if obj is None else obj
 
+def opt_str_float_param(obj: Any, param_name: str, default: str = None) -> Optional[str]:
```

Comment:
This seems pretty specific... my preference would be to just do the isinstance check directly in the AWS library than adding a function to the `dagster.check` module.

Reply:
Validation of parameters datatype is happening in dagster.check module only so I have added this.

Reply:
ok got it @prha .